### PR TITLE
Align font features

### PR DESCRIPTION
**Pull request by Brooooooklyn**
> [!NOTE]
> Adds broad Canvas2D text/font support aligned with W3C/WHATWG and MDN examples.
> 
> - New properties: `textRendering`, `lang`, `fontOpticalSizing`, `fontFeatureSettings`, `fontVariantLigatures`, `fontVariantNumeric`, `fontVariantPosition`, `fontSizeAdjust` (getters/setters, state, typings)
> - Skia integration: passes explicit features, applies `opsz` (auto), honors variable axes (`wght`/`wdth` unless overridden), computes `font-size-adjust` via x-height, locale-aware ligatures (e.g., Turkish disables `fi`), and `textRendering` toggles kerning/ligatures
> - Parsing: support `font-feature-settings` syntax; improve `font` shorthand parsing (reject global keywords, default `oblique` 14deg with clamping, round numeric weights, disallow negative sizes, handle percentage as size)
> - API surface: updates `index.d.ts` for new types/fields; adds FFI structs (`FontFeature`) and extends C++/Rust bridges and render paths to carry features/options
> - Tests: extensive unit and snapshot tests for kerning, ligatures, numeric forms, superscripts/subscripts, optical sizing, size-adjust, text-rendering modes, locale effects, and parsing edge cases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 343be29f6e9b1c15bf2b67316cb081bd94f7191e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: skia-c/skia_c.cpp [hunk 1]
```diff
@@ -507,6 +507,12 @@ void skiac_canvas_get_line_metrics_or_draw_text(
     }
   }
 
+  // Apply font weight as 'wght' variation for variable fonts
+  // 'wght' tag = 0x77676874
+  if (weight != 400) {
+    coords.push_back({SkSetFourByteTag('w', 'g', 'h', 't'), static_cast<float>(weight)});
```
Comment (yisibl) [inline skia-c/skia_c.cpp]: Let me do it at #1162.

File: skia-c/skia_c.cpp [hunk 2]
```diff
@@ -554,6 +647,38 @@ void skiac_canvas_get_line_metrics_or_draw_text(
     text_style.addFontFeature(SkString("titl"), 1);
   }
 
+  // Apply language/locale for language-specific glyph variants
+  // lang: BCP-47 language tag (e.g., "en", "tr", "zh-Hans") or
```
Comment (yisibl) [inline skia-c/skia_c.cpp:651]: 按照 Chrome 开发者给出的答复，这里我们不需要兼容 BCP-47 算法，交给底层的 harfbuzz 来处理就好了

File: src/font.rs [hunk 3]
```diff
@@ -127,6 +146,10 @@ impl Font {
             // Try to parse optional angle
             if let Ok(angle) = parser.try_parse(parse_oblique_angle) {
               font.oblique_angle = Some(angle);
+            } else {
+              // Per W3C CSS Fonts Level 4: default oblique angle is 14deg
+              // See: https://drafts.csswg.org/css-fonts-4/#font-style-prop
+              font.oblique_angle = Some(14.0);
```
Comment (yisibl) [inline src/font.rs:152]: font-style 需要重写，这块也比较复杂，单独来做吧。

File: index.d.ts [hunk 4]
```diff
@@ -199,19 +199,51 @@ type CanvasFontVariantCaps =
...
   direction: CanvasDirection
   font: string
   fontKerning: CanvasFontKerning
   fontStretch: CanvasFontStretch
   fontVariantCaps: CanvasFontVariantCaps
+  fontOpticalSizing: CanvasFontOpticalSizing
+  fontVariantLigatures: CanvasFontVariantLigatures
+  fontVariantNumeric: CanvasFontVariantNumeric
+  fontVariantPosition: CanvasFontVariantPosition
+  fontSizeAdjust: number | null
```
Comment (yisibl) [inline index.d.ts:238]: 让我们标识出哪些是目前 canvas 规范中的，哪些是我们实验性的私有方法。
